R/step_epi_shift.R

```diff
@@ -229,22 +216,46 @@ bake.step_epi_shift <- function(object, new_data, ...) {
 }
 
 #' @export
-print.step_epi_lag <- function(x, ...) {
-  NextMethod(title = "Lagging ", shift = abs(x$shift))
+bake.step_epi_ahead <- function(object, new_data, ...) {
+  grid <- tidyr::expand_grid(col = object$columns, ahead = object$ahead) %>%
+    dplyr::mutate(newname = glue::glue("{object$prefix}{ahead}_{col}"),
+                  shift_val = -object$ahead,
+                  ahead = NULL)
+
+  ## ensure no name clashes
+  new_data_names <- colnames(new_data)
+  intersection <- new_data_names %in% grid$newname
+  if (any(intersection)) {
+    rlang::abort(
+      paste0("Name collision occured in `", class(object)[1],
+             "`. The following variable names already exists: ",
+             paste0(new_data_names[intersection], collapse = ", "),
+             "."))
+  }
+  ok <- object$keys
+  shifted <- reduce(
+    pmap(grid, epi_shift_single, x = new_data, key_cols = ok),
+    dplyr::full_join,
+    by = ok
+  )
+
+  dplyr::full_join(new_data, shifted, by = ok) %>%
+    dplyr::group_by(dplyr::across(dplyr::all_of(ok[-1]))) %>%
+    dplyr::arrange(time_value) %>%
+    dplyr::ungroup()
+
 }
 
+
 #' @export
-print.step_epi_ahead <- function(x, ...) {
-  NextMethod(title = "Leading ", shift = abs(x$shift))
+print.step_epi_lag <- function(x, width = max(20, options()$width - 30), ...) {
+  print_step_shift(x$columns, x$terms, x$trained, "Lagging ", shift = x$lag)
+  invisible(x)
 }
 
 #' @export
-print.step_epi_shift <- function(x,
-                                 width = max(20, options()$width - 30),
-                                 title = "Shifting ",
-                                 shift = x$shift,
-                                 ...) {
-  print_step_shift(x$columns, x$terms, x$trained, title, width, shift = shift)
+print.step_epi_ahead <- function(x, width = max(20, options()$width - 30), ...) {
+  print_step_shift(x$columns, x$terms, x$trained, "Leading ", shift = x$ahead)
   invisible(x)
 }
```

Comment:
Where does `print_step_shift` come from?

Reply:
If you click the little blue arrow just 👇 , it should expand the script.

If the code is already on the target branch, it won't be shown in the PR. This function isn't changed by the PR.